### PR TITLE
Use New func to protect race condition for cloudevent client.

### PR DESCRIPTION
**Pull request by n3wscott**
Followup to #248 

**Release Note**

```release-note
NONE
```

/assign @vaikas-google

### CODE REVIEW

File: cmd/github_receive_adapter/main.go [hunk 1]
```diff
@@ -56,8 +56,9 @@ func main() {
 
 	log.Printf("Sink is: %q", *sink)
 
-	ra := &github.Adapter{
-		SinkURI: *sink,
+	ra, err := github.New(*sink)
+	if err != nil {
+		log.Fatalf("Failed to create github adaptor: %s", err.Error())
```
Comment (evankanderson) [inline cmd/github_receive_adapter/main.go]: I think you want "adapter": https://writingexplained.org/adapter-vs-adaptor-difference

("Adapter" is overwhelmingly more popular in American English; British English is more evenly split, but there are no specific rules differentiating the two.)

Comment (n3wscott) [reply cmd/github_receive_adapter/main.go]: every time...

Comment (n3wscott) [reply cmd/github_receive_adapter/main.go]: fixed.

**Pull request merged**